source/agora/protocol/Cache.d

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+
+    Contains the Cache.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.protocol.Cache;
+
+/// Use to cache messages.
+struct Cache(T, U)
+{   
+private:
+    U[T] base;
```

Comment:
This module is just reimplementing a hashmap. You can just uses `get` and `in` and `update` in hashmap.

Reply:
Modified.

Reply:
@Geod24